File: projects/sales/src/lib/models/name.ts
export class Name {

  constructor(
    public title: string = '',
    public givenName: string = '',
    public middleName: string = '',
    public familyName: string = '',
    public preferredName: string = '',
    public initials: string = '',
    public honorific: string = '',
    public salutation: string = ''

  ) {}

}

/*

  private String title;         // name prefix
  private String givenName;
  private String preferredGivenName;
  private String middleName;    // otherNames
  private String familyName;
  private String preferredFamilyName;
  private String preferredName; // informalSalutation
  private String initials;
  private String honorific;     // name suffix
  private String salutation;    // formalSalutation

*/

File: backend/modules/party-service/src/main/java/org/serendipity/party/entity/Name.java
package org.serendipity.party.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Embeddable;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Embeddable
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Setter
@Getter
public class Name {

  private String title;

  private String givenName;

  private String preferredName;

  private String middleName;

  @Column(name = "familyName", nullable = false)
  private String familyName;

  private String initials;

  private String honorific;

  private String salutation;

}

// name prefix
// informalSalutation
// otherNames
// name suffix
// formalSalutation
